include/libdnf/comps/environment/environment.hpp

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_COMPS_ENVIRONMENT_ENVIRONMENT_HPP
+#define LIBDNF_COMPS_ENVIRONMENT_ENVIRONMENT_HPP
+
+#include "libdnf/base/base_weak.hpp"
+#include "libdnf/transaction/transaction_item_reason.hpp"
+
+#include <set>
+#include <string>
+#include <vector>
+
+
+namespace libdnf::comps {
+
+
+struct EnvironmentId {
+public:
+    EnvironmentId() = default;
+    explicit EnvironmentId(int id) : id(id) {}
+
+    bool operator==(const EnvironmentId & other) const noexcept { return id == other.id; }
+    bool operator!=(const EnvironmentId & other) const noexcept { return id != other.id; }
+
+    // Corresponds to solvable id
+    int id{0};
+};
+
+
+/// @replaces dnf:dnf/comps.py:class:Environment
+class Environment {
+public:
+    const std::vector<EnvironmentId> & get_environment_ids() const { return environment_ids; }
+    void add_environment_id(EnvironmentId environment_id) { environment_ids.push_back(environment_id); }
+
+    /// Get environment id
+    std::string get_environmentid() const;
```

Comment:
I think both the methods that expose the libsolv ids can be removed (`get_environment_ids` and `add_environment_id`). It is used only in `EnvironmentQuery`, but the query can access the `environment_ids` directly. It would also make the interface better, even though there would still be the private `environment_ids` variable. Does it sound good?

Reply:
I think it does, though I'm not sure if the IDs would be needed (or just useful to have) on the API. The decision to have them or not on the API should be based on that, not on us having a naming problem :slightly_smiling_face:

Reply:
The naming problem was just a side note. It makes sense to me to not have them on the interface. And it would be easier to add them later when we find out it's needed than vice versa.